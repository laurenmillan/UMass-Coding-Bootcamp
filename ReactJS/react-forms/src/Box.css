.Box {
	text-align: center;
	display: 'inline-block';
	margin: '10px';
}
