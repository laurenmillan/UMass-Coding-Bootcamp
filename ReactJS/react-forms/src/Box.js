import React from 'react';
import './Box.css';

/** Displays colored box.
 * 
 * Props: width, height, color, removeBox
 * 
 * - this component should display a div with a background color, 
 *      width and height based on the props passed to it.
 * - When each Box component is displayed, add a button with the text of of “X” next to each Box. 
 * - When this button is clicked, remove that specific box. 
 * - This will require you to pass a function down as props - the button should not be a separate component, 
 *      it should be included in the Box component.
 *
 **/

const Box = ({ width, height, color, removeBox }) => {
	const boxStyle = {
		backgroundColor: color,
		width: `${width}px`,
		height: `${height}px`
	};

	return (
		<div style={boxStyle}>
			<button onClick={removeBox}>X</button>
		</div>
	);
};

export default Box;
