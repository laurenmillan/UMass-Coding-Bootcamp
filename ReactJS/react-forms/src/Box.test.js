import React from 'react';
import { render } from '@testing-library/react';
import Box from './Box';

// SMOKE TEST
describe('Box', () => {
	test('it renders without crashing', () => {
		render(<Box />);
	});
});

// SNAPSHOT TEST
test('it matches snapshot', () => {
	const { asFragment } = render(<Box />);
	expect(asFragment()).toMatchSnapshot();
});
