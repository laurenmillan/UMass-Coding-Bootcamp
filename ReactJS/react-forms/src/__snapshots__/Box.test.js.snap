// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="display: inline-block; margin: 10px;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
